src/snowflake/connector/pandas_tools.py

```diff
@@ -389,7 +394,7 @@ def drop_object(name: str, object_type: str) -> None:
         target_table_location = build_location_helper(
             database,
             schema,
-            random_string() if overwrite else table_name,
+            random_string() if (overwrite and auto_create_table) else table_name,
```

Comment:
This seems like an unrelated bug fix to what you describe in your PR description. Could you write a test for this?

Reply:
it is related to the bug fix (and the bug itself), but nonetheless I wrote a new test for it (since I hadn't written earlier)